### PR TITLE
Remove legacy code for DELETE and UPDATE

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# SPI
* Remove `ConnectorMetadata` methods `getDeleteRowIdColumnHandle()`, `beginDelete()`, `finishDelete()`,
  `getUpdateRowIdColumnHandle()`, `beginUpdate()`, and `finishUpdate()`. ({issue}`15161`)
* Remove interface `UpdatablePageSource`. ({issue}`15161`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/metadata/Metadata.java [hunk 1]
```diff
@@ -362,26 +352,6 @@ Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
...
-
-    /**
-     * Finish delete query
-     */
-    void finishDelete(Session session, TableHandle tableHandle, Collection<Slice> fragments);
-
-    /**
-     * Begin update query
-     */
-    TableHandle beginUpdate(Session session, TableHandle tableHandle, List<ColumnHandle> updatedColumns);
```
Comment (nineinchnick) [inline core/trino-main/src/main/java/io/trino/metadata/Metadata.java:378]: The `beginMerge` method doesn't have a list of updated columns, so `beginMerge` can't replace it in all cases. I opened #15585 with more details about my use case so it's easier to discuss it here, but IMO we should resolve it before merging this PR.

**Comment by kokosing** [comment]
Can you please rebase?

**Comment by findepi** [comment]
I allowed myself to do a rebase.

**Comment by electrum** [comment]
@findepi @kokosing rebased, should be good this time

**Review by dain — APPROVED** [review]
Looks good

**Pull request merged**